game/campaignloader/defaultsquadronassigner.py

```diff
@@ -26,7 +26,7 @@ def __init__(
         self.coalition = coalition
         self.air_wing = coalition.air_wing
         self.squadron_defs = SquadronDefLoader(game, coalition).load()
-        self.squadron_def_generator = SquadronDefGenerator(self.coalition)
+        self.air_wing.squadron_def_generator = SquadronDefGenerator(self.coalition)
```

Comment:
Create this in the `AirWing` constructor?

Reply:
i tried to already but something was not right, i will have a look again. I do not like to initialize this as well out of the constructor. Leads to some problem otherwise. But i took the trade because it happens during the Game Generation logic which is like a constructor for the game. Will have a closer look

Reply:
(TIL, submit draft comments before clicking "unresolve", because otherwise GitHub deletes your draft 😠)

> Will have a closer look

Unresolving so we don't forget to revisit then :)

If this does for some reason need to be initialized outside the constructor, the `AirWing` constructor needs to initialize the member to `None`. Declaring `squadron_def_generator: SquadronDefGenerator` at the class level is a lie to the type checker. That's telling mypy that `AirWing` has a _class_ variable named `squadron_def_generator`, but it's never actually defined. Any attempt to access `AirWing.squadron_def_generator` between the time of construction and this line will raise `AttributeError`, but the type system doesn't know that. You want `self.squadron_def_generator: Optional[SquadronDefGenerator] = None` in `AirWing.__init__` if this is the behavior you need.

Reply:
> (TIL, submit draft comments before clicking "unresolve", because otherwise GitHub deletes your draft 😠)

I know what you mean!

Already moved the initialization to the right place: Inside the AirWing constructor. Needed some changes as the SquadronDefGenerator requires the faction data but that was a small change. Thats why i resolved it already :)